tests/core/math/test_projection.h

```diff
@@ -0,0 +1,807 @@
+	Vector2 actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							 .get_far_plane_half_extents();
+
+	CHECK(expected.x == doctest::Approx(actual.x));
+	CHECK(expected.y == doctest::Approx(actual.y));
+}
+
+TEST_CASE("[Projection] get_fov") {
+	real_t expected = 64.8575;
+	real_t actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							.get_fov();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] flipped_y") {
+	Projection expected(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87));
+	Projection actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(-55.01, 55.50, -87.92, -36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+								.flipped_y();
+
+	CHECK(expected.is_equal_approx(actual));
+}
+
+TEST_CASE("[Projection] get_aspect") {
+	real_t expected = 0.218863;
+	real_t actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							.get_aspect();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_far_plane_half_extents") {
+	Vector2 expected = Vector2(0.200337, 0.158066);
+	Vector2 actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							 .get_far_plane_half_extents();
+
+	CHECK(expected.x == doctest::Approx(actual.x));
+	CHECK(expected.y == doctest::Approx(actual.y));
+}
+
+TEST_CASE("[Projection] get_fov") {
+	real_t expected = 64.8575;
+	real_t actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+							.get_fov();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_fovy") {
+	real_t expected = 1.32997;
+	real_t actual = Projection().get_fovy(1.0, 1.33);
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_lod_multiplier") {
+	float expected = 1;
+	float actual = Projection().get_lod_multiplier();
+
+	CHECK(expected == doctest::Approx(actual));
+}
+
+TEST_CASE("[Projection] get_pixels_per_meter") {
+	int expected = 55;
+	int actual = Projection(
+			Vector4(-29.32, 43.39, -94.85, -89.95),
+			Vector4(55.01, -55.50, 87.92, 36.67),
+			Vector4(48.63, 19.80, 75.48, 44.12),
+			Vector4(-2.16, 70.96, -59.63, 23.87))
+						   .get_pixels_per_meter(64);
```

Comment:
```suggestion
						 .get_pixels_per_meter(64);
```

Looks like clang format isn't happy with this whitespace. I just copy-pasted the suggestion from the failing CI